contracts/votingMachines/GenesisProtocolLogic.sol

```diff
@@ -0,0 +1,652 @@
+        _;
+    }
+
+    /**
+     * @dev register a new proposal with the given parameters. Every proposal has a unique ID which is being
+     * generated by calculating keccak256 of a incremented counter.
+     * @param _paramsHash parameters hash
+     * @param _proposer address
+     * @param _organization address
+     */
+    function propose(uint , bytes32 _paramsHash,address _proposer,address _organization)
+        external
+        returns(bytes32)
+    {
+        //Check parameters existence.
+        require(parameters[_paramsHash].quedVoteRequiredPercentage > 0);
+            // Generate a unique ID:
+        bytes32 proposalId = keccak256(abi.encodePacked(this, proposalsCnt));
+        proposalsCnt++;
+         // Open proposal:
+        Proposal memory proposal;
+        proposal.callbacks = msg.sender;
+        proposal.organizationId = keccak256(abi.encodePacked(msg.sender,_organization));
+
+        proposal.state = ProposalState.Qued;
+        // solium-disable-next-line security/no-block-members
+        proposal.times[0] = now;
+        proposal.currentBoostedVotePeriodLimit = parameters[_paramsHash].boostedVotePeriodLimit;
+        proposal.proposer = _proposer;
+        proposal.winningVote = NO;
+        proposal.paramsHash = _paramsHash;
+        if (organizations[proposal.organizationId] == 0) {
+            if (_organization == address(0)) {
+                organizations[proposal.organizationId] = msg.sender;
+            } else {
+                organizations[proposal.organizationId] = _organization;
+            }
+        }
+        //calc dao bounty
+        uint daoBounty = parameters[_paramsHash].daoBountyConst.mul(averagesBoostDownstakes[proposal.organizationId]);
+        if (daoBounty < parameters[_paramsHash].minimumDaoBounty) {
+            proposal.daoBountyRemain = parameters[_paramsHash].minimumDaoBounty;
+        } else {
+            proposal.daoBountyRemain = daoBounty;
+        }
+        proposal.totalStakes = proposal.daoBountyRemain;
+        proposals[proposalId] = proposal;
+        proposals[proposalId].stakes[NO] = proposal.daoBountyRemain;//dao downstake on the proposal
+        emit NewProposal(proposalId, organizations[proposal.organizationId], NUM_OF_CHOICES, _proposer, _paramsHash);
+        return proposalId;
+    }
+
+    /**
+      * @dev expired try to execute a boosted proposal if it is expired
+      * @param _proposalId the id of the proposal
+      * @return uint expirationCallBounty the bounty amount for the expiration call
+     */
+    function expired(bytes32 _proposalId) external returns(uint expirationCallBounty) {
+        Proposal storage proposal = proposals[_proposalId];
+        require(proposal.state == ProposalState.Boosted);
+        require(_execute(_proposalId));
+        // solium-disable-next-line security/no-block-members
+        uint expirationCallBountyPercentage = (1 + now.sub(proposal.currentBoostedVotePeriodLimit + proposal.times[1]).div(15));
```

Comment:
the formula is (1+t)% where t is number of blocks pass from the proposal expiration time.
because we are currently not counting blocks (we are counting seconds) so 15 is a bit more than average block time.. we might need this value (15) as a param.
It is capped by 100%.
How it can negative ?

Reply:
if  proposal.tiems[1] = 1, now = 2 and currentBoostedVotePriodLimit = 99 (i.e. when the boosted period has not finished yet), it'll be negative. (I do not see where/if you are checking that the period is actually expired...)

Reply:
proposal.tiems[1]+ currentBoostedVotePriodLimit is the expiration time of the proposal ...so it is always > now

Reply:
Ok, so I think I get it. The code:

- checks if the prop is boosted
- executes it successfully

And because Boosted proposals can only be executed after they expired, you can conclude that this date < now. 

I'm not a big fan of this (it is very easy to change the (very complex) _execute code in a later iteratino and introduce a bug here, but for the rest +1 :-)